CMakeLists.txt

```diff
@@ -292,6 +292,10 @@ if (NOT USE_SIMD STREQUAL "")
             else ()
                 add_definitions ("-m${feature}")
             endif ()
+            if (feature STREQUAL "fma" AND CMAKE_COMPILER_IS_CLANG)
+                # Clang needs an explicit flag to make fma ops
+                add_definitions ("-ffp-contract=fast")
```

Comment:
It *IS* opt-in, it only is triggered by building with USE_SIMD=fma, which does not happen by default.

Note that it's only clang that needs this fp-contract business. For gcc, once -mfma is specified on the command line, the compiler will automatically turn a*b+c into fma instructions. So presumably, smarter people than us have already grappled with this and decided it's safe.

As a matter of fact, I was fooling around with some time trials this weekend in the presence of -mfma, which is how this patch came about. Most of our workstations at work don't have the hardware to support it, but on my laptop it showed appreciable differences for certain benchmarks. Enough to make me wish we had fma everywhere so we could enable it for production builds.